helm/resource_release.go

```diff
@@ -454,21 +475,24 @@ func resourceReleaseCreate(d *schema.ResourceData, meta interface{}) error {
 
 	values, err := getValues(d)
 	if err != nil {
-		return err
+		return diag.FromErr(err)
 	}
 
-	validInstallableChart, err := isChartInstallable(chart)
+	validInstallableChart, err := isChartInstallable(c)
 	if !validInstallableChart {
-		return err
+		if err != nil {
+			return diag.FromErr(err)
+		}
+		return diag.FromErr(errors.New("Unknown error. isChartInstallable returned 'false'"))
```

Comment:
I don't think we actually need this diagnostic, if you look at the code for `isChartInstallable` it will always return an error with `false` so this code path will never get hit. We could just refactor that function to only return an `error`.